src/platform/packages/private/kbn-esql-editor/src/helpers.ts

```diff
@@ -63,7 +63,10 @@ export const useDebounceWithOptions = (
   );
 };
 
-const quotedWarningMessageRegexp = /"(.*?)"/g;
+// Quotes can be used as separators for multiple warnings unless
+// they are escaped with backslashes. This regexp will match any
+// quoted string that is not escaped.
+const quotedWarningMessageRegexp = /(?<!\\)"(.*?)(?<!\\)"/g;
```

Comment:
I think we can avoid lookbehind and backtracking with the following: 
```suggestion
const quotedWarningMessageRegexp = /"(?:[^"\\]*(?:\\.[^"\\]*)*)"/g;
```

Reply:
@darnautov thanx, done https://github.com/elastic/kibana/pull/213685/commits/a81274c0ea0dc13771d645c22c50c7fd68bc90b4